src/inmanta/server/services/environmentservice.py

```diff
@@ -0,0 +1,293 @@
+"""
+    Copyright 2019 Inmanta
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+    Contact: code@inmanta.com
+"""
+import asyncio
+import logging
+import os
+import shutil
+import time
+import uuid
+from typing import List, Optional, cast
+
+import asyncpg
+
+from inmanta import const, data
+from inmanta.protocol import encode_token, methods
+from inmanta.protocol.common import attach_warnings
+from inmanta.protocol.exceptions import NotFound, ServerError
+from inmanta.server import (
+    SLICE_AGENT_MANAGER,
+    SLICE_DATABASE,
+    SLICE_ENVIRONMENT,
+    SLICE_ORCHESTRATION,
+    SLICE_RESOURCE,
+    SLICE_SERVER,
+    protocol,
+)
+from inmanta.server.agentmanager import AgentManager
+from inmanta.server.server import Server
+from inmanta.server.services.orchestrationservice import OrchestrationService
+from inmanta.server.services.resourceservice import ResourceService
+from inmanta.types import Apireturn, JsonType, Warnings
+
+LOGGER = logging.getLogger(__name__)
+
+
+class EnvironmentService(protocol.ServerSlice):
+    """Slice with project and environment management"""
+
+    server_slice: Server
+    agentmanager: AgentManager
+    orchestration_service: OrchestrationService
+    resource_service: ResourceService
+
+    def __init__(self) -> None:
+        super(EnvironmentService, self).__init__(SLICE_ENVIRONMENT)
+
+    def get_dependencies(self) -> List[str]:
+        return [SLICE_SERVER, SLICE_DATABASE, SLICE_AGENT_MANAGER]
+
+    async def prestart(self, server: protocol.Server) -> None:
+        self.server_slice = cast(Server, server.get_slice(SLICE_SERVER))
+        self.agentmanager = cast(AgentManager, server.get_slice(SLICE_AGENT_MANAGER))
+        self.orchestration_service = cast(OrchestrationService, server.get_slice(SLICE_ORCHESTRATION))
```

Comment:
SLICE_ORCHESTRATION and SLICE_RESOURCE are used here but they are not part of the dependency list

Reply:
done